test/functional/schema_migrations.js

```diff
@@ -178,4 +213,169 @@ describe('Schema migration', function() {
                     'error message looks wrong');
         });
     });
+
+    it('handles adding static columns', function() {
+        var newSchema = clone(testTable0);
+        newSchema.version = 6;
+        newSchema.attributes.added_static_column = 'string';
+        newSchema.index = [{ attribute: 'added_static_column', type: 'static' }].concat(newSchema.index);
```

Comment:
Should we care about the order of the static column relative to the partition keys? Logically, it would probably make most sense to order it after the 'hash' entries, but I'm not sure if we should enforce or merely encourage that for consistency.